server/routes/apis/topicRoutes.js

```diff
@@ -5,14 +5,64 @@
  * see included LICENSE or https://opensource.org/licenses/BSD-3-Clause 
  */
 
-const express = require( 'express' );
-const router = express.Router( );
-
-// import controllers
-const topicController = require( '../../controller/apis/topicController' );
-
-
-
-
-
-module.exports = router;
\ No newline at end of file
+ const express = require( 'express' );
+ const router = express.Router( );
+  
+ // import controllers
+ const topicController = require( '../../controller/apis/topicController' );
+  
+ /*
+  * Topics can be requested the following ways
+  * / <- all visible Topics for the auth user (all Topics a user can see, owned, shared with user, or set to public visibility)
+  * /user <- all Topics for the user (does not include additional shared or visible ones)
+  * /shared <- all Topics that are shared with the user but not their own
+  * /visible <- all Topics that are publicly visible
+  * /sharedAndVisible <- all Topics that are shared or visible to the user but are not their own
+  */ 
+ 
+ 
+ router.route( '/' )
+     .get(async function ( req, res ) {
+         topicController.getAllVisibleTopics( req, res );
+     })    
+     .post( ( req, res ) => { 
+         topicController.saveTopic( req, res );
+     }
+ )
+ 
+ /**
+  * Returns all active Topics owned by the user
+  */
+ router.route( '/user/:id' )
+     .get( async ( req, res ) => {
+         topicController.getAllActiveTopicsForUser( req, res );
+     }
+ )
+ 
+ router.route( '/shared' )
+     .get( async ( req, res ) => {
+         topicController.getAllSharedTopicsForUser( req, res );
+     })
+ 
+ // Topics /api/v1/auth/topics/:id
+ router.route( '/:id' )
+     .get( async function ( req, res ) {
+         topicController.getTopicById( req, res );
+     
+     })
+     .delete( async ( req, res ) => {
+         topicController.deleteTopicById( req, res );
```

Comment:
I know our convention has been to use deleteById, but I think it's worth having a conversation about this. Should delete be handled this way? Or should it be handled similarly to save on the '/' route, passing the object in req.body?

Reply:
I agree this could be another possible approach. We should definitely have a discussion about this sometime in the near future.

Reply:
Agreed. We should have a conversation about this.